### ISSUE TITLE
growth-chart-app

### ISSUE THREAD

**Issue by alpborcek** [smart-on-fhir/smart-dev-sandbox#11]
Hi I am a doctor with no significant knowledge on code and servers. I need to test the growth-chart-app. I applied every instruction but I can not reach the app. Can you please guide me or lead me to a guide on how to lunch a smart on fhir app on a local machine. I have ubuntu - windows - mac and virtualbox installed on each platform but I could not be successfull

**Comment by vlad-ignatov** [comment]
Hi,

You've posted the issue here, so I assume you have the local sandbox running. The next thing to do is to get the Growth Chart app (you need to have Git and NodeJS installed):

```sh
git clone https://github.com/smart-on-fhir/growth-chart-app.git
cd growth-chart-app
npm i
npm start
```
At this point you should see a message in the terminal, saying that the app is running on certain url (E.g.`http://localhost:8000`). This means that your launchUrl would be http://localhost:8000/launch.html

Then go to http://localhost:4000 and click on "Launch a SMART App". Enter the app launch URL  select a patient that is not older then 20 yrs and has vital-sign observations and launch.

Note that you **DO NOT** need to run a local sandbox to try the app locally (unless you want to edit the patient data). Here is a link that will launch the PGC app agains the online sandbox:

https://launch.smarthealthit.org/?auth_error=&fhir_version_1=r4&fhir_version_2=r4&iss=&launch_ehr=1&launch_url=https%3A%2F%2Fexamples.smarthealthit.org%2Fgrowth-chart-app%2Flaunch.html&patient=17ff80a3-dda8-4a58-a8c5-6fe94ad0d747&prov_skip_auth=1&provider=&pt_skip_auth=1&public_key=&sb=&sde=&sim_ehr=0&token_lifetime=15&user_pt=

You can use that and only change the `App Launch URL` field to point to your local app (e.g. http://localhost:8000)

**Comment by alpborcek** [comment]
Thank you so much for the help I really appreciate 

I am trying on a mac and everything is going fine until I try to run the app. Then it says(http://127.0.0.1:9000/ is my url)  "No 'state' parameter found. Please (re)launch the app." I also recieve two npm WARNs 1-No repository field and 2 - No lcience field. And nothing goes on from there. 

Actually I want to learn the details on lunching the app over a linux web server. Is there a way to setup a sandbox in a web server or can I do it without a sandbox.

**Comment by vlad-ignatov** [comment]
Hi @alpborcek,

The smart-dev-sandbox can be used in any OS, including linux. That gives you the sandbox environment (FHIR servers, data, launcher etc). The app that you want to launch is a separate entity. For convenience, `npm start` starts a small server that will make it available at http://127.0.0.1:9000/, but you can also "host it yourself using a server like Apache or NginX.

It seems to me that we have a different issue here. This is a SMART app that **CAN NOT** be launched directly (by visiting the url in the browser). If the app is available at http://127.0.0.1:9000/, then your launch url is http://127.0.0.1:9000/launch.html and your redirect URL is http://127.0.0.1:9000/. You can use [this link](https://launch.smarthealthit.org/?auth_error=&fhir_version_1=r4&fhir_version_2=r4&iss=&launch_ehr=1&launch_url=http%3A%2F%2F127.0.0.1%3A9000%2Flaunch.html&patient=17ff80a3-dda8-4a58-a8c5-6fe94ad0d747&prov_skip_auth=1&provider=&pt_skip_auth=1&public_key=&sb=&sde=&sim_ehr=0&token_lifetime=15&user_pt=) to try it. You can also register a client at the HSPC Sandbox if you find that easier.

See [this](http://www.hl7.org/fhir/smart-app-launch/) for details on how SMART apps are launched.
